libbpf-rs/src/print.rs

```diff
@@ -34,27 +56,33 @@ extern "C" fn outer_print_cb(
     fmtstr: *const c_char,
     va_list: *mut libbpf_sys::__va_list_tag,
 ) -> i32 {
-    // can't be null: we cant't get here until print_cb has been initialized with a function
-    // pointer.
-    let cb = PRINT_CB.load(Ordering::Relaxed);
-    let cb = unsafe { std::mem::transmute::<*const (), PrintCallback>(cb) };
+    let cb = from_ptr(PRINT_CB.load(Ordering::Relaxed));
     match unsafe { vsprintf::vsprintf(fmtstr, va_list) } {
```

Comment:
Some callbacks may filter by log level (eg benchmark tests may discard all debug logs). In those cases, it'd be great to just return early here instead of calling vsprintf and calling into the cb. Maybe in the `set_print` API, we could also take in the level to filter by as another parameter, and then in this function here, check whether the message should be discarded or forwarded to the cb based off that filter. What are your thoughts on this?

Reply:
That's a good point. Rather than complicate the API, I changed the paramter from `&str` to `&dyn Display`. This way it will be lazily evaluated only when the user needs it.